### DISCUSSION TITLE
App does not always 'connect' properly.

### DISCUSSION THREAD

**Discussion by GijsTimmers** [BimmerGestalt/AAIdrive#227] [Q&A]
I will add screenshots. The app does not always connect properly and it will ask me to reconnect bluetooth or connect via USB. I don't know what triggers this behaviour.

**Comment by hufman** [comment]
This state appears when AndroidAutoIdrive detects that the car is connected properly, but also that BMW Connected hasn't started the tunnel to the car. This is a limitation in BMW Connected, unfortunately.
The instructions to reconnect Bluetooth are the best way to trigger the tunnel to start working, I believe. I don't know a good way to keep BMW Connected always ready to notice the first Bluetooth connection, but the second one almost always works.

**Comment by bogdan-calapod** [comment]
What I have done is make a Bixby Routine to automatically open the BMW Connected app when the car connects via Bluetooth. It still borks out sometimes, but it generally works much better than without this. 😁

**Comment by berseker** [comment]
I noticed that after every update, it is better to strictly recheck the full appliance of the tips mentioned here for both AAI & Bmw\Mini Connected in order to reduce the chances to have issues with phone connectivity  https://hufman.github.io/AndroidAutoIdrive/connection.html

**Comment by kalahcustom** [comment]
I achieve the best 'connect' accuracy if I keep Connected app in 'lock' mode so it is never closed. I was using bixby routine, but app connected properly only about 70% cases. Mostly it didn't trigger Spotify on time and needed to reconnect Bluetooth. 
While using the 'lock' method I have never faced initial connection problem again. 
![Screenshot_20210311-221241_One UI Home](https://user-images.githubusercontent.com/77300479/110855723-089c0200-82b7-11eb-8305-e2b53a5512e3.jpg)
